zds/forum/commons.py

```diff
@@ -100,21 +92,24 @@ class PostEditMixin(object):
     @staticmethod
     def perform_hide_message(request, post, user, data):
         is_staff = user.has_perm('forum.change_post')
-        if post.author == user or is_staff:
-            for alert in post.alerts_on_this_comment.all():
-                alert.solve(user, _(u'Le message a été masqué.'))
-            post.is_visible = False
-            post.editor = user
-
-            if is_staff:
-                post.text_hidden = data.get('text_hidden', '')
-
-            messages.success(request, _(u'Le message est désormais masqué.'))
-            for user in Notification.objects.get_users_for_unread_notification_on(post):
-                signals.content_read.send(sender=post.topic.__class__, instance=post.topic, user=user)
-        else:
+
+        if not post.author == user and not is_staff:
             raise PermissionDenied
 
+        for alert in post.alerts_on_this_comment.all():
+            alert.solve(user, _(u'Le message a été masqué.'))
+
+        post.is_visible = False
```

Comment:
d'ailleurs avoirs une méthode de `utils.models.comment` qui fait :  

```python
def hide(self, save=False, user=None, moderation_message=''):
    self.solve_attached_alerts()
    self.is_visible=False
    self.text_hidden=moderation_message
    for user in Notification.objects.get_users_for_unread_notification_on(post):
        signals.content_read.send(sender=post.topic.__class__, instance=post.topic, user=user)
 ```

comme ça tu n'aurais plus qu'à appeler `post.hide(user=user, moderation_message=data.get('text_hidden', is_staff and ''))`

Reply:
À quoi servirait le paramètre `save` ? J'ai modifié en conséquence de tes commentaires.